client/CServerHandler.h

```diff
@@ -0,0 +1,112 @@
+﻿/*
+ * CServerHandler.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#include "../lib/CStopWatch.h"
+
+#include "../lib/StartInfo.h"
+
+struct SharedMemory;
+class CConnection;
+class PlayerColor;
+struct StartInfo;
+
+struct ServerCapabilities;
+class CMapInfo;
+struct ClientPlayer;
+class CMapHeader;
+struct LobbyGuiAction;
+struct PlayerInfo;
+struct CPackForLobby;
+
+template<typename T> class CApplier;
+class CBaseForLobbyApply;
+
+class IServerAPI
+{
+public:
+	virtual void sendClientConnecting() =0;
+	virtual void sendClientDisconnecting() =0;
+	virtual void setMapInfo(std::shared_ptr<CMapInfo> to, CMapGenOptions * mapGenOpts = nullptr) =0;
+	virtual void setPlayer(PlayerColor color) =0;
+	virtual void setPlayerOption(ui8 what, ui8 dir, PlayerColor player) =0;
+	virtual void setDifficulty(int to) =0;
+	virtual void setTurnLength(int npos) =0;
+	virtual void sendMessage(const std::string & txt) =0;
+	virtual void sendGuiAction(ui8 action) =0; // TODO: possibly get rid of it?
+	virtual void sendStartGame() =0;
+};
+
+class CClient; //MPTODO: rework
+/// structure to handle running server and connecting to it
+class CServerHandler : public IServerAPI, public LobbyInfo
+{
+	SharedMemory * shm;
+
+	CApplier<CBaseForLobbyApply> * applier;
+
+	boost::recursive_mutex * mx;
+	std::list<CPackForLobby *> incomingPacks; //protected by mx
+
+	CStopWatch th;
```

Comment:
naming by shortcuts is not the best idea, it's good but for small ranges, I would change these names

Reply:
I generally agree with this, but since code is still quite a mess and there no proper getters I feel like things such as ```si```, ```mi``` that used very often is reasonable trade off until another refactoring.